service/src/main/java/org/jboss/sbomer/service/feature/sbom/features/generator/AbstractController.java

```diff
@@ -211,7 +211,7 @@ protected boolean isFinished(TaskRun taskRun) {
     protected Boolean isSuccessful(TaskRun taskRun) {
         if (!isFinished(taskRun)) {
             log.trace("TaskRun '{}' still in progress", taskRun.getMetadata().getName());
-            return null;
+            return null; // FIXME: This is not really binary, but trinary state
```

Comment:
Sonar is complaining about all these comparisons that do not use a primitive type.